test/dotcom/utils/service_date_time_test.exs

```diff
@@ -0,0 +1,240 @@
+defmodule Dotcom.Utils.ServiceDateTimeTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  import Dotcom.Utils.ServiceDateTime
+
+  import Test.Support.Generators.DateTime
+  import Test.Support.Generators.ServiceDateTime
+
+  describe "service_rollover_time/0" do
+    test "returns a time for the service rollover time" do
+      # Exercise/Verify
+      assert %Time{} = service_rollover_time()
+    end
+  end
+
+  describe "service_date/1" do
+    property "returns 'today' when the date_time is between the start of service and midnight" do
+      # Setup
+      check all(date_time <- date_time_generator(:before_midnight)) do
+        beginning_of_service_day = beginning_of_service_day(date_time)
+        end_of_day = Timex.end_of_day(date_time)
+
+        date_time_generator =
+          time_range_date_time_generator({beginning_of_service_day, end_of_day})
+
+        check all(service_date_time <- date_time_generator) do
+          # Exercise
+          date = Timex.to_date(date_time)
+          service_date = service_date(service_date_time)
+
+          # Verify
+          assert Timex.equal?(date, service_date, :day)
+        end
+      end
+    end
+
+    property "returns 'yesterday' when the date_time is between midnight and the end of service" do
+      # Setup
+      check all(date_time <- date_time_generator(:after_midnight)) do
+        beginning_of_day = Timex.end_of_day(date_time) |> Timex.shift(microseconds: 1)
+        end_of_service_day = end_of_service_day(date_time)
+
+        date_time_generator =
+          time_range_date_time_generator({beginning_of_day, end_of_service_day})
+
+        check all(service_date_time <- date_time_generator) do
+          yesterday = service_date_time |> Timex.shift(days: -1)
+
+          # Exercise
+          date = Timex.to_date(yesterday)
+          service_date = service_date(date_time)
+
+          # Verify
+          assert Timex.equal?(date, service_date, :day)
+        end
+      end
+    end
+  end
+
+  describe "service_range/1" do
+    test "returns :past for past" do
+      # Setup
+      today = beginning_of_service_day()
+      past = Timex.shift(today, microseconds: -1)
+
+      # Exercise/Verify
+      assert service_range(past) == :past
+    end
+
+    test "returns :today for today" do
+      # Setup
+      today = service_range_day() |> random_time_range_date_time()
+
+      # Exercise/Verify
+      assert service_range(today) == :today
+    end
+
+    test "returns :this_week for this week" do
+      # Setup
+      {_, end_of_current_service_week} = service_range_current_week()
```

Comment:
I alluded to this in [a different comment](https://github.com/mbta/dotcom/pull/2372/files#r1949908609), but given that all of the assertions about "current week" also use the implementation of `service_range_current_week()`, I don't have much visibility into what the test is actually asserting.

Reply:
I just changed to the test to accept `:today` or `:this_week` since we don't really have an easy way to pick a day that isn't a specific day.

Reply:
All of our functions except this one *do* take a date_time and use now as a default. The reason this one doesn't is because it's entire purpose is to give you an atom that expresses the relation of the date_time you do pass to a range of time relative to now. Avoiding mocking isn't a good strategy, but, the question is whether it is worth mocking Timex in order to test this one function when we can be pretty sure it is behaving as we expect by reading the code. There are tradeoffs.